### ISSUE TITLE
W0621 not enforced

### ISSUE THREAD

**Issue by chrisspen** [pylint-dev/pylint#8923]
### Bug description

Run:
    $ cd /tmp
    c$ python3.11 -m venv .env
    c$ . .env/bin/activate
    (.env) c$ pip install pylint
    Collecting pylint
      Downloading pylint-2.17.5-py3-none-any.whl (536 kB)
         ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 536.7/536.7 kB 2.8 MB/s eta 0:00:00
    Collecting platformdirs>=2.2.0 (from pylint)
      Downloading platformdirs-3.10.0-py3-none-any.whl (17 kB)
    Collecting astroid<=2.17.0-dev0,>=2.15.6 (from pylint)
      Downloading astroid-2.15.6-py3-none-any.whl (278 kB)
         ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 278.3/278.3 kB 16.5 MB/s eta 0:00:00
    Collecting isort<6,>=4.2.5 (from pylint)
      Downloading isort-5.12.0-py3-none-any.whl (91 kB)
         ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 91.2/91.2 kB 9.0 MB/s eta 0:00:00
    Collecting mccabe<0.8,>=0.6 (from pylint)
      Using cached mccabe-0.7.0-py2.py3-none-any.whl (7.3 kB)
    Collecting tomlkit>=0.10.1 (from pylint)
      Downloading tomlkit-0.12.1-py3-none-any.whl (37 kB)
    Collecting dill>=0.3.6 (from pylint)
      Downloading dill-0.3.7-py3-none-any.whl (115 kB)
         ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 115.3/115.3 kB 10.8 MB/s eta 0:00:00
    Collecting lazy-object-proxy>=1.4.0 (from astroid<=2.17.0-dev0,>=2.15.6->pylint)
      Downloading lazy_object_proxy-1.9.0-cp311-cp311-manylinux_2_5_x86_64.manylinux1_x86_64.manylinux_2_17_x86_64.manylinux2014_x86_64.whl (64 kB)
         ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 64.8/64.8 kB 2.8 MB/s eta 0:00:00
    Collecting wrapt<2,>=1.14 (from astroid<=2.17.0-dev0,>=2.15.6->pylint)
      Downloading wrapt-1.15.0-cp311-cp311-manylinux_2_5_x86_64.manylinux1_x86_64.manylinux_2_17_x86_64.manylinux2014_x86_64.whl (78 kB)
         ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 78.9/78.9 kB 8.4 MB/s eta 0:00:00
    Installing collected packages: wrapt, tomlkit, platformdirs, mccabe, lazy-object-proxy, isort, dill, astroid, pylint
    Successfully installed astroid-2.15.6 dill-0.3.7 isort-5.12.0 lazy-object-proxy-1.9.0 mccabe-0.7.0 platformdirs-3.10.0 pylint-2.17.5 tomlkit-0.12.1 wrapt-1.15.0

    [notice] A new release of pip is available: 23.1.2 -> 23.2.1
    [notice] To update, run: pip install --upgrade pip
    (.env) c$ cat > test.py << EOL
    > #/usr/bin/env python
    > import math as _
    > class MyClass:
    >     def mymeth(self):
    >         _ = 123
    >         print(_.abs(-123))
    > EOL
    $ pylint test.py
    ************* Module test
    test.py:1:0: C0114: Missing module docstring (missing-module-docstring)
    test.py:3:0: C0115: Missing class docstring (missing-class-docstring)
    test.py:4:4: C0116: Missing function or method docstring (missing-function-docstring)
    test.py:6:14: E1101: Instance of 'int' has no 'abs' member (no-member)
    test.py:3:0: R0903: Too few public methods (1/2) (too-few-public-methods)

Notice that  the sample Python redefines the global `_` inside the method. This should throw a [W0621](https://pylint.readthedocs.io/en/latest/user_guide/messages/warning/redefined-outer-name.html) error, but does not.

### Configuration

```ini
NA/Defaults
```

### Command used

```shell
See above.
```

### Pylint output

```shell
See above.
```

### Expected behavior

W0621 should be enforced.

### Pylint version

```shell
2.17.5
```

### OS / Environment

Ubuntu 20

**Comment by mbyrnepr2** [comment]
There's a default regex which underpins if the redefined-outer-name message is emitted or not:
```python
"_+$|(_[a-zA-Z0-9_]*[a-zA-Z0-9]+?$)|dummy|^ignored_|^unused_"
```

So, any variable starting with (or equalling) `_` leads to this and that's part of the design currently.

I wonder would we be going in the right direction by expecting users to re-use the `_` variables in code?

**Comment by chrisspen** [comment]
Yeah, I submitted this because I ran into a bug where a dev used `_` to ignore unused values in a tuple returned by a function, and unknowingly overrode the Django `_` translation function. e.g.

    value, _, _ in self.get_values()

Don't know why you'd want to exclude variables prefixed with `_` from this rule check.

**Comment by mbyrnepr2** [comment]
@chrisspen I imagine the rationale is the same reason `unused-variable` also depends on the same regex: often the `_` variables are not intended to be used, they are "throwaway" as in the example you provided.
That's not to say there isn't a problem, since `_` is a valid name and isn't always necessarily throwaway, as is the case here.

So, looks like a breaking change would be required, to provide a separate regex configuration for the two checkers.
